api/envoy/config/core/v3/protocol.proto

```diff
@@ -53,8 +53,37 @@ message QuicKeepAliveSettings {
   }];
 }
 
+// Config for QUIC connection migration across network interfaces, i.e. cellular to WIFI, upon
+// network change events from the platform, i.e. the current network gets
+// disconnected, or upon the QUIC detecting a bad connection. After migration, the
+// connection may be on a different network other than the default network
+// picked by the platform. A connection is this special state is only allowed to
+// serve new requests for a certain period of time before being drained, and
+// meanwhile, QUIC will try to migrate to the default network if possible.
+message ConnectionMigrationSettings {
```

Comment:
If this is QUIC only settings (and will not be used in other places), I think it should be inside the `QuicProtocolOptions`).